OpenLibraClient/docs/man/man3/Book.3
.TH "Book" 3 "Sat Sep 3 2011" "Version 1.0.0" "OpenLibra Client" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Book \- 
.SH SYNOPSIS
.br
.PP
.PP
\fC#import <Book.h>\fP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "(id) - \fBinitWithDictionary:\fP"
.br
.in -1c
.SS "Static Public Member Functions"

.in +1c
.ti -1c
.RI "(\fBBook\fP *) + \fBbookWithDictionary:\fP"
.br
.in -1c
.SS "Properties"

.in +1c
.ti -1c
.RI "int \fBid\fP"
.br
.ti -1c
.RI "NSString * \fBtitle\fP"
.br
.ti -1c
.RI "NSString * \fBauthor\fP"
.br
.ti -1c
.RI "NSString * \fBpublisher\fP"
.br
.ti -1c
.RI "int \fBpublisherDate\fP"
.br
.ti -1c
.RI "int \fBpages\fP"
.br
.ti -1c
.RI "NSString * \fBlanguage\fP"
.br
.ti -1c
.RI "NSString * \fBURLDetails\fP"
.br
.ti -1c
.RI "NSString * \fBURLDownload\fP"
.br
.ti -1c
.RI "NSString * \fBcover\fP"
.br
.ti -1c
.RI "float \fBrating\fP"
.br
.ti -1c
.RI "int \fBnumComments\fP"
.br
.ti -1c
.RI "NSString * \fBcategories\fP"
.br
.ti -1c
.RI "NSString * \fBtags\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
Class to represent a \fBBook\fP in the OpenLibra service.
.PP
Books class stores all properties of a book and some helper methods to download, view or comment the book 
.SH "Member Function Documentation"
.PP 
.SS "+ (\fBBook\fP *) bookWithDictionary: (NSDictionary *)dictionary"Creates an autoreleased book using given dictionary
.PP
\fBReturns:\fP
.RS 4
\fBBook\fP Autoreleased book 
.RE
.PP

.SS "- (id) initWithDictionary: (NSDictionary *)dictionary"Intialices a new \fBBook\fP object using given NSDictionary to fill all properties
.PP
\fBParameters:\fP
.RS 4
\fINSDictionary\fP Dictionary with all properties in a key/value format 
.RE
.PP
\fBReturns:\fP
.RS 4
\fBBook\fP 
.RE
.PP

.SH "Property Documentation"
.PP 
.SS "- (NSString* Book.)author\fC [read, write, retain]\fP"Book's author 
.SS "- (NSString* Book.)categories\fC [read, write, retain]\fP"Book's categories 
.SS "- (NSString* Book.)cover\fC [read, write, retain]\fP"String with url to book's cover 
.SS "- (int Book.)id\fC [read, write, assign]\fP"Unique identifier of the book 
.SS "- (NSString* Book.)language\fC [read, write, retain]\fP"Book's language 
.SS "- (int Book.)numComments\fC [read, write, assign]\fP"Number of comments about this book 
.SS "- (int Book.)pages\fC [read, write, assign]\fP"Book's pages 
.SS "- (NSString* Book.)publisher\fC [read, write, retain]\fP"Book's publisher 
.SS "- (int Book.)publisherDate\fC [read, write, assign]\fP"Book's publisher date 
.SS "- (float Book.)rating\fC [read, write, assign]\fP"Book's rating 
.SS "- (NSString* Book.)tags\fC [read, write, retain]\fP"Book's tags 
.SS "- (NSString* Book.)title\fC [read, write, retain]\fP"Book's title 
.SS "- (NSString* Book.)URLDetails\fC [read, write, retain]\fP"String with the url to book's details 
.SS "- (NSString* Book.)URLDownload\fC [read, write, retain]\fP"String with the download url 

.SH "Author"
.PP 
Generated automatically by Doxygen for OpenLibra Client from the source code.


OpenLibraClient/OpenLibraClient/Book.m
//  This file is part of the OpenLibraClient package.
//
//  Book.h
//  OpenLibraClient
//
//  Created by Víctor Berga on 21/08/11 <victor@victorberga.com>.
//  Copyright 2011 Víctor Berga. All rights reserved.
// 
//  For the full copyright and license information, please view the LICENSE
//  file that was distributed with this source code.

#import "Book.h"

NSString * const kIdKey             = @"ID";
NSString * const kTitleKey          = @"title";
NSString * const kAuthorKey         = @"author";
NSString * const kPublisherKey      = @"publisher";
NSString * const kPublisherDateKey  = @"publisher_date";
NSString * const kPagesKey          = @"pages";
NSString * const kLanguageKey       = @"language";
NSString * const kURLDetailsKey     = @"url_details";
NSString * const kURLDownloadKey    = @"url_download";
NSString * const kCoverKey          = @"cover";
NSString * const kRatingKey         = @"rating";
NSString * const kNumCommentsKey    = @"num_comments";
NSString * const kCategoriesKey     = @"categories";
NSString * const kTagsKey           = @"tags";


@implementation Book

@synthesize id              = _id;
@synthesize title           = _title;
@synthesize author          = _author;
@synthesize publisher       = _publisher;
@synthesize publisherDate   = _publisherDate;
@synthesize pages           = _pages;
@synthesize language        = _language;
@synthesize URLDetails      = _URLDetails;
@synthesize URLDownload     = _URLDownload;
@synthesize cover           = _cover;
@synthesize rating          = _rating;
@synthesize numComments     = _numComments;
@synthesize categories      = _categories;
@synthesize tags            = _tags;

- (id)initWithDictionary:(NSDictionary *)dictionary
{
    self = [super init];
    if (self) {
        _id             = [[dictionary valueForKey:kIdKey] intValue];
        _title          = [dictionary valueForKey:kTitleKey];
        _author         = [dictionary valueForKey:kAuthorKey];
        _publisher      = [dictionary valueForKey:kPublisherKey];
        _publisherDate  = [[dictionary valueForKey:kPublisherDateKey] intValue];
        _pages          = [[dictionary valueForKey:kPagesKey] intValue];
        _language       = [dictionary valueForKey:kLanguageKey];
        _URLDetails     = [dictionary valueForKey:kURLDetailsKey];
        _URLDownload    = [dictionary valueForKey:kURLDownloadKey];
        _cover          = [dictionary valueForKey:kCoverKey];
        _rating         = [[dictionary valueForKey:kRatingKey] floatValue];
        _numComments    = [[dictionary valueForKey:kNumCommentsKey] intValue];
        _categories     = [dictionary valueForKey:kCategoriesKey];
        _tags           = [dictionary valueForKey:kTagsKey];
    }
    
    return self;
}

#pragma mark -
#pragma mark Memory Management

- (void)dealloc
{
    [self setId:0];
    [self setTitle:nil];
    [self setAuthor:nil];
    [self setPublisher:nil];
    [self setPublisherDate:0];
    [self setPages:0];
    [self setLanguage:nil];
    [self setURLDetails:nil];
    [self setURLDownload:nil];
    [self setCover:nil];
    [self setRating:0.0];
    [self setNumComments:0];
    [self setCategories:nil];
    [self setTags:nil];
    
    [super dealloc];
}

#pragma mark -
#pragma mark Class Methods

+ (Book *)bookWithDictionary:(NSDictionary *)dictionary
{
    return [[[Book alloc] initWithDictionary:dictionary] autorelease];
}

#pragma mark -
#pragma mark Debug methods

- (NSString *)description
{
    return [NSString stringWithFormat:@"Book {\
            id: %d \
            title: %@ \
            author: %@ \
            publisher: %@ \
            publisher_date: %d \
            pages: %d \
            url_details: %@ \
            url_downloads: %@ \
            cover: %@ \
            rating: %f \
            num_comments: %d \
            categories: %@ \
            tags: %@",
            self.id, self.title, self.author, self.publisher, 
            self.publisherDate, self.pages, self.URLDetails,
            self.URLDownload, self.cover, self.rating, self.numComments,
            self.categories, self.tags];
}

@end
